File: AndroidLearnProject/app/src/main/code_backup/mpresenter/TodoAddPresenter.java
package com.android.learn.mpresenter;


import com.android.learn.base.mmodel.BaseData;
import com.android.learn.base.mpresenter.BasePresenter;
import com.android.learn.base.thirdframe.rxjava.BaseObserver;
import com.android.learn.base.utils.ResponseStatusUtil;
import com.android.learn.mcontract.TodoAddContract;
import com.android.learn.mcontract.TodoEditContract;

import java.util.Map;

import io.reactivex.Observable;


public class TodoAddPresenter extends BasePresenter<TodoAddContract.View> implements TodoAddContract.Presenter {

    @Override
    public void addTodo( Map<String, Object> map) {
        Observable observable = mRestService.addTodo(map);

        addSubscribe(observable, new BaseObserver<BaseData>(false) {

            @Override
            public void onNext(BaseData data) {
                if (data.errorCode == BaseData.SUCCESS) {
                    mView.showAddTodo(data);
                } else ResponseStatusUtil.handleResponseStatus(data);
            }
        });
    }

}

File: AndroidLearnProject/app/src/main/java/com/android/learn/mpresenter/TodoAddPresenter.kt
package com.android.learn.mpresenter


import com.android.learn.base.mmodel.BaseData
import com.android.learn.base.mpresenter.BasePresenter
import com.android.learn.base.thirdframe.rxjava.BaseObserver
import com.android.learn.base.utils.ResponseStatusUtil
import com.android.learn.mcontract.TodoAddContract
import com.android.learn.mcontract.TodoEditContract

import io.reactivex.Observable


class TodoAddPresenter : BasePresenter<TodoAddContract.View>(), TodoAddContract.Presenter {

    override fun addTodo(map: Map<String, Any>) {
        val observable = mRestService.addTodo(map)

        addSubscribe(observable, object : BaseObserver<BaseData>(false) {

            override fun onNext(data: BaseData) {
                if (data.errorCode == BaseData.SUCCESS) {
                    mView!!.showAddTodo(data)
                } else
                    ResponseStatusUtil.handleResponseStatus(data)
            }
        })
    }

}
